### ISSUE TITLE
[no-assign-mutated-array] doesn't account for `flatMap`

### ISSUE THREAD

**Issue by SamuelT-Beslogic** [cartant/eslint-plugin-etc#69]
The rule will accept sorting a mapped array, but not a flatMapped one.
I suppose the rule checks for known function names, `flatMap` is common enough is should probably be excluded.

OK (but [unicorn/prefer-array-flat-map](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v46.0.1/docs/rules/prefer-array-flat-map.md))
```ts
const biggestQuantity = entityFilters
  .map(category => category.items.map(item => item.quantity))
  .flat(1)
  .sort((quantity1, quantity2) => quantity2 - quantity1)
```

OK (invalid code, just for linting demonstration purposes)
```ts
const biggestQuantity = entityFilters
  .map(category => category.items.map(item => item.quantity))
  .sort((quantity1, quantity2) => quantity2[0] - quantity1[0])
```

Fails `no-assign-mutated-array`
```ts
const biggestQuantity = entityFilters
  .flatMap(category => category.items.map(item => item.quantity))
  .sort((quantity1, quantity2) => quantity2 - quantity1)
```
